### PR TITLE
Pretty-printing uses CRLF on Windows. Closes #14808.

### LINKED ISSUE #14808: Pretty printing should use \r\n on Windows.
On Windows the preferred line ending is \r\n, not \n like it is on posix platforms.
Unfortunately using --pretty with rustc outputs \n newlines.
For comparison clang-format outputs \r\n on Windows.

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: src/libsyntax/print/pp.rs [hunk 1]
```diff
@@ -507,8 +507,14 @@ impl<'a> Printer<'a> {
         }
     }
     pub fn print_newline(&mut self, amount: isize) -> io::Result<()> {
+        #[cfg(windows)]
+        const NEWLINE: &'static str = "\r\n";
+
+        #[cfg(not(windows))]
+        const NEWLINE: &'static str = "\n";
```
Comment (alexcrichton) [inline src/libsyntax/print/pp.rs]: Could you use `cfg!` here  instead of `#[cfg]` as well? (helps weed out compile errors on all platforms at once)

Comment (tamird) [inline src/libsyntax/print/pp.rs]: Do you mean:
```diff
diff --git a/src/libsyntax/print/pp.rs b/src/libsyntax/print/pp.rs
index 15aaf9c..c0d68ef 100644
--- a/src/libsyntax/print/pp.rs
+++ b/src/libsyntax/print/pp.rs
@@ -507,8 +507,13 @@ impl<'a> Printer<'a> {
         }
     }
     pub fn print_newline(&mut self, amount: isize) -> io::Result<()> {
+        let newline = if cfg!(windows) {
+            "\r\n"
+        } else {
+            "\n"
+        };
         debug!("NEWLINE {}", amount);
-        let ret = write!(self.out, "\n");
+        let ret = self.out.write_all(newline.as_bytes());
         self.pending_indentation = 0;
         self.indent(amount);
         return ret;
```
?

Comment (alexcrichton) [inline src/libsyntax/print/pp.rs]: Yeah that'd do the trick! You could also use `b"\n"` to avoid the call to `.as_bytes()`

Comment (tamird) [inline src/libsyntax/print/pp.rs]: Done!

Comment (tamird) [inline src/libsyntax/print/pp.rs]: Gah, that doesn't typecheck. Sec

**Comment by tamird** [comment]
OK, this should work.

**Comment by klutzy** [comment]
I'm against this change. If the output depends on OS, it leads to subtle bugs which can't be fixed by non-Windows users.
Also, we already have backlog of various CRLF bugs: e.g. #7723, #8731, and #12471. These bugs are sometimes hard to fix (or even hard to figure out what's wrong).
Actually [the test failed](https://github.com/rust-lang/rust/pull/24696#issuecomment-95480417) during `run-make` due to extra `\r`.

**Comment by azurespace** [comment]
I also think this change must not be merged without enough concern. If this feature is needed for some situation, it should not be solved such a way. It would be more suitable solution to add an environmental variable EVERY platform as a flag to enable/disable.

**Comment by tamird** [comment]
I'm fine with this not going in - I was just looking to close #14808. Perhaps that issue should be closed in favour of an RFC?

**Comment by alexcrichton** [comment]
Sure, sounds like there's not consensus on this issue. I don't think this necessarily needs an RFC, but it would probably be best to reach consensus on the issue first before pursuing this. Thanks regardless though @tamird!

**Pull request closed without merging**